scene/main/canvas_item.cpp

```diff
@@ -500,6 +523,7 @@ void CanvasItem::_top_level_changed() {
 			child->_top_level_changed_on_parent();
 		}
 	}
+	update_configuration_warnings();
```

Comment:
I can move this back to `set_as_top_level` if there would be performance concerns calling it down a deep tree and add a note to this as well, works either way